docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,334 @@
+        reduce:
+            type: {"1D, "3D"}
+            stacked: {true, false}
+            background: {"irf", "reflected", "ring"}
+            roi: max_offset
+            exclusion: exclusion.fits
+        fit:
+            energy_min, energy_max
+        logging:
+            level: debug
+
+    grid:
+        spatial: center, width, binsz
+        energy: min, max, nbins
+        time: min, max
+        # PSF RAD and EDISP MIGRA not exposed for now
+        # Per-obs energy_safe and roi_max ?
+    
+    observations:
+        data_store: $GAMMAPY_DATA/cta-1dc/index/gps/
+        ids: 110380, 111140, 111159
+        conesearch: 
+            ra:
+            dec:
+            radius:
+            energy_min:
+            energy_max:
+            time_min:
+            time_max:
+
+    model:
+        # Model configuration will mostly be designed in different PIG
+        sources:
+            source_1:
+                spectrum: powerlaw
+                spatial: shell
+            diffuse: gal_diffuse.fits
+        background:
+            IRF
+
+**API design**
+
+The design of the high-level interface API is driven by the use cases considered, the different
+tools (tasks) identified and their responsibilities, as well as the need of a main ``Analysis``
+session object that drives and orchestrates internally the different tools involved, their inputs
+and products. The ``Analysis`` session object will be initialized with a configuration file (see *prototype
+configuration file* above) and will be the responsible to instantiate and run the different tools
+classes (see *session workflow* below). The tools are middle  management agents (e.g. MapMaker,
+ReflectedBgEstimator, ...) responsible to perform the different tasks identified in the use cases covered.
+
+The ``Analysis`` session object will provide access to every object involved and data structure produced
+during the session. ``Analysis`` methods calls will produce and modify datasets (i.e. models, maps,..),
+but in between method calls advanced users can do a lot of custom processing by their own with scripting
+using the Gammapy Python toolbox.
+
+The code of the ``Analysis`` class as well as any other eventual class needed will be placed in ``gammapy.scripts``.
+This module will contain also the set of different command line tools provided, where some small cleaning
+and refactoring may be needed (i.e. remove ``gammapy image`` command line tool)
+
+**Serialisation**
+
+The user could choose in the configuration file to serialise the products delivered by the tools 
+with the help of a boolean parameter. The serialisation would then be performed as follows:
+
+- Every tool in Gammapy knows how to serialise/read themselves
+- Composite objects will be serialised as a series of all of their components
+- Serialisation will be a mix of YAML (models, state) and FITS files (maps)
+- There will not be framework supporting different serialisation backends, we will have one way.
+
```

Comment:
I do not understand what it means....

Reply:
different backends...

Reply:
@Bultako - My suggestion would be to reduce, not extend the infos on serialisation.
I don't think we can work out the solution for that in this PIG.
It would require it's own PIG, and partly will be covered by the data reduction PIG that others want to write in the coming month or two, and partly that work is already starting, see e.g. #2314.
If we couple this PIG to that work, we will delay it by 1-2 months, but really this PIG just says "we plan to implement a config-file based high-level interface for Gammapy" as a plan and we should aim to limit this PIGs scope and have it approved ASAP.

Reply:
Ok
I've simplified (and also I hope clarified) the paragraph on serialisation.